docs/sources/flow/reference/components/faro.receiver.md

```diff
@@ -0,0 +1,267 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/components/faro.receiver/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/faro.receiver/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/faro.receiver/
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/faro.receiver/
+title: faro.receiver
+---
+
+# faro.receiver
+
+`faro.receiver` accepts web application telemetry data from the [Grafana Faro Web SDK][faro-sdk]
+and forwards it to other components for future processing.
+
+[faro-sdk]: https://github.com/grafana/faro-web-sdk
+
+## Usage
+
+```river
+faro.receiver "LABEL" {
+    output {
+        logs   = [LOGS_RECEIVERS]
+        traces = [OTELCOL_COMPONENTS]
+    }
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`extra_log_labels` | `map(string)` | Extra labels to attach to emitted log lines. | `{}` | no
+
+## Blocks
+
+The following blocks are supported inside the definition of `faro.receiver`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+server | [server][] | Configures the HTTP server. | no
+server > rate_limiting | [rate_limiting][] | Configures rate limiting for the HTTP server. | no
+sourcemaps | [sourcemaps][] | Configures sourcemap retrieval. | no
+sourcemaps > location | [location][] | Configures on-disk location for sourcemap retrieval. | no
+output | [output][] | Configures where to send collected telemetry data. | yes
+
+[server]: #server-block
+[rate_limiting]: #rate_limiting-block
+[sourcemaps]: #sourcemaps-block
+[location]: #location-block
+[output]: #output-block
+
+### server block
+
+The `server` block configures the HTTP server managed by the `faro.receiver`
+component. Clients using the [Grafana Faro Web SDK][faro-sdk] forward telemetry
+data to this HTTP server for processing.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`listen_address` | `string` | Address to listen for HTTP traffic on. | `127.0.0.1` | no
+`listen_port` | `number` | Port to listen for HTTP traffic on. | `12347` | no
+`cors_allowed_origins` | `list(string)` | Origins for which cross-origin requests are permitted. | `[]` | no
+`api_key` | `secret` | Optional API key to validate client requests with. | `""` | no
+`max_allowed_payload_size` | `number` | Maximum size (in bytes) for client requests. | `5e6` | no
```

Comment:
nit: In the future we can have this use unit2bytes like we do for other components. Let's just leave a TODO comment for it.

Reply:
I'll change that now; shouldn't be too much effort.